src/compiler/emitter.ts

```diff
@@ -848,8 +848,8 @@ export function emitFiles(resolver: EmitResolver, host: EmitHost, targetSourceFi
         const filesForEmit = forceDtsEmit ? sourceFiles : filter(sourceFiles, isSourceFileNotJson);
         // Setup and perform the transformation to retrieve declarations from the input files
         const inputListOrBundle = compilerOptions.outFile ? [factory.createBundle(filesForEmit)] : filesForEmit;
-        if (emitOnly && !getEmitDeclarations(compilerOptions)) {
-            // Checker wont collect the linked aliases since thats only done when declaration is enabled.
+        if ((emitOnly && !getEmitDeclarations(compilerOptions)) || compilerOptions.noCheck) {

```

Comment:
Also need to add forceDtsEmit check here? shouldnt signature generation behave like `noCheck`?

Reply:
AFAIK it already does - the `|| compilerOptions.noCheck` being tacked on here is so that `noCheck` builds do the same data collection signature generation runs already do.